include/rocksdb/memory_allocator.h

```diff
@@ -36,9 +36,40 @@ class MemoryAllocator {
   }
 };
 
+namespace jemalloc {
+
+enum class PerCPUArena : int {
+  // Disable per-CPU arena. One single arena is used.
+  kDisabled = 0,
+
+  // Create one arena for each CPU, to reduce arena mutex contention,
+  // but may increase memory fragmentation.
+  kPerCPU = 1,
+
+  // Assume hyper-threading is enabled and create one arena for each physical
+  // CPU (half of #CPU in total). The allocator does not have run-time check to 
+  // verify whether the assumption is true. This mode may have less memory
+  // fragmentation comparing to kPerCPU, for using less arenas.
+  kPerPhysicalCPU = 2,
```

Comment:
Do we need to provide options at all? If we only recommend users to use one setting, then we should not provide an option for now until there is a use case.

Reply:
They are handful for me to run benchmarks. Plus these may provide trade-off between mutex contention and memory fragmentation. If you like I can hide them though.

Reply:
Be aware that they are user facing. Any more option to user introduces potential costs going forward. Can you remove all other options after your benchmark?

Reply:
Let's also benchmark with even fewer arenas, e.g. using nCPU / 4 or 8, since tcache is enabled I'd think they should be fine.

The default jemalloc setting optimizes scalability since small allocations like <64B happen very often. IIRC the popular workload here should be on the order of KB in general? If so scaling the # of arenas should be fine.

Reply:
Just add to another perspective. It's not uncommon for RocksDB users to have multiple processes running RocksDB. In one setting, 8-12 processes are running. I'm not sure it is safe and desirable to have each one to have nCPU number of arenas, or we should allow users to create fewer arenas.